src/functionalTest/groovy/com/autonomousapps/android/AdviceSpec.groovy

```diff
@@ -32,6 +32,9 @@ final class AdviceSpec extends AbstractAndroidSpec {
           onUnusedAnnotationProcessors {
               fail()
           }
+          onAdvisedCompileOnly {
```

Comment:
I think `onCompileOnly` is good enough.

Reply:
And as far as the test is concerned, I think that you want two compileOnly dependencies on the classpath. Then you can have this:
```groovy
onCompileOnly {
  fail("...") // one of the compileOnly identifiers
}
```
and finally, the test assertion should be that the list of compileOnly deps that are in the advice should not include the one being here filtered.